projects/solar/test/test_fan_control_solar.c

```diff
@@ -0,0 +1,234 @@
+  };
+
+  // Ensure fan_control_init works
+  TEST_ASSERT_OK(fan_control_init(&settings));
+
+  for (Mppt i = 1; i <= 6; i++) {
+    data_store_set(DATA_POINT_TEMPERATURE(i), 2);
+  }
+  data_store_done();
+
+  // Overtemperature fault should not be raised
+  MS_TEST_HELPER_ASSERT_NO_EVENT_RAISED();
+
+  // Full Speed pin should be high to disable full speed
+  gpio_get_state(&s_full_speed_addr, &s_gpio_state);
+  TEST_ASSERT_EQUAL(GPIO_STATE_HIGH, s_gpio_state);
+}
+
+// Case 2: Temperature above threshold for one MPPT
+void test_fan_overtemp_fault_fullspeed_handling_case_oneabovetemp(void) {
+  FanControlSettingsSolar settings = {
+    .overtemp_addr = s_overtemp_addr,
+    .fan_fail_addr = s_fan_fail_addr,
+    .full_speed_addr = s_full_speed_addr,
+    .full_speed_temp_threshold = s_full_speed_temp_threshold,
+    .mppt_count = s_mppt_count,
+  };
+  TEST_ASSERT_OK(fan_control_init(&settings));
+
+  for (Mppt i = 1; i <= 5; i++) {
+    data_store_set(DATA_POINT_TEMPERATURE(i), 2);
+  }
+  data_store_set(DATA_POINT_TEMPERATURE(6), 4);
+  data_store_done();
+
+  // Overtemperature fault should be raised
+  TEST_ASSERT_OK(event_process(&s_e));  // Make sure an event exists
+  // Should fail by default if event_process(&e) fails
+  MS_TEST_HELPER_ASSERT_EVENT(s_e, EE_SOLAR_FAULT_FAN_OVERTEMPERATURE, 0);
+
+  // Full Speed pin should be low to enable full speed
+  gpio_get_state(&s_full_speed_addr, &s_gpio_state);
+  TEST_ASSERT_EQUAL(GPIO_STATE_LOW, s_gpio_state);
+}
+
+// Case 3: Temperature above threshold for multiple MPPTs
+void test_fan_overtemp_fault_fullspeed_handling_case_multipleabovetemp(void) {
+  FanControlSettingsSolar settings = {
+    .overtemp_addr = s_overtemp_addr,
+    .fan_fail_addr = s_fan_fail_addr,
+    .full_speed_addr = s_full_speed_addr,
+    .full_speed_temp_threshold = s_full_speed_temp_threshold,
+    .mppt_count = s_mppt_count,
+  };
+  TEST_ASSERT_OK(fan_control_init(&settings));
+
+  for (Mppt i = 1; i <= 6; i++) {
+    data_store_set(DATA_POINT_TEMPERATURE(i), 5);
+  }
+  data_store_done();
+
+  // Overtemperature fault should be raised
+  TEST_ASSERT_OK(event_process(&s_e));  // Make sure an event exists
+  MS_TEST_HELPER_ASSERT_EVENT(s_e, EE_SOLAR_FAULT_FAN_OVERTEMPERATURE, 0);
+
+  // Full Speed pin should be low to enable full speed
+  gpio_get_state(&s_full_speed_addr, &s_gpio_state);
+  TEST_ASSERT_EQUAL(GPIO_STATE_LOW, s_gpio_state);
+}
+
+// Case 4: Overtemperature status on any MPPT should set full speed pin low
+void test_fan_overtemp_fault_fullspeed_handling_case_overtempstatus(void) {
+  FanControlSettingsSolar settings = {
+    .overtemp_addr = s_overtemp_addr,
+    .fan_fail_addr = s_fan_fail_addr,
+    .full_speed_addr = s_full_speed_addr,
+    .full_speed_temp_threshold = s_full_speed_temp_threshold,
+    .mppt_count = s_mppt_count,
+  };
+  TEST_ASSERT_OK(fan_control_init(&settings));
+
+  // Set mppt(6)'s status to overtemperature
+  data_store_set(DATA_POINT_MPPT_STATUS(6), 2);  // "0000010"
```

Comment:
Aside from this needing to be zero indexed, you should also use `SPV1020_OVT_MASK` (OVT == overtemperature) from spv1020_mppt_defs.h instead of the raw number